### PR TITLE
Feat/application form

### PR DESCRIPTION
**Pull request by FearsomeRover**

### CODE REVIEW

File: apps/frontend/src/components/ui/ApplicationForm.tsx [hunk 1]
```diff
@@ -58,24 +61,41 @@ const formSchema = z
...
+    }
+  }, [user.data]);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async function onSubmit({ terms, ...values }: z.infer<typeof formSchema>) {
     try {
-      const response = await axios.post('/api/application', values);
-      if (response.status === 200) {
+      const updateResponse = await api.patch('/users/me', values);
+      const response = await api.post('/application', {
```
Comment (mozsarmate) [inline apps/frontend/src/components/ui/ApplicationForm.tsx:95]: the user is able to send their application even if they fail to update their data?

**Comment by FearsomeRover** [comment]
Yea, we need to think about what the correct order of business here is. I think first we need to update the profile data, and if it is successful onl then post the application. Also do we need to have some sort of confirmation when the users room hasn't changed? Or do they only need to check the rules checkbox then?

**Comment by mozsarmate** [comment]
Yeah, i agree on this, although your code does not check whether the profile data update was successful or not.
I don't think we would need another checkbox, would say that the opportunity to change your room number is quite clear.

**Comment by FearsomeRover** [comment]
> Yeah, i agree on this, although your code does not check whether the profile data update was successful or not. I don't think we would need another checkbox, would say that the opportunity to change your room number is quite clear.

Well axios throws error, if the response status is 4-500 so the application wont be submitted.

**Review by mozsarmate — APPROVED** [review]

**Pull request merged**